hibernate-core/src/main/java/org/hibernate/id/uuid/UuidGenerator.java

```diff
@@ -81,6 +83,12 @@ private static UuidValueGenerator determineValueGenerator(
 			else if ( config.style() == TIME ) {
 				return new CustomVersionOneStrategy();
 			}
+			else if ( config.style() == VERSION_6 ) {
+				return new UuidVersion6Strategy();
+			}
+			else if ( config.style() == VERSION_7 ) {
```

Comment:
Not sure if this will look any better than now ...

```
if ( config.algorithm() != UuidValueGenerator.class ) {
...
}
else {
    switch (config.style()) {
    ...
    }
}
```

Reply:
You won't need the `else` since the first if statement returns.

Reply:
Or like

```java
if(config == null) {
  return StandardRandomStrategy.INSTANCE;
}
if(config.algorithm() != UuidValueGenerator.class) {
  // throw or return logic
}

switch(config.style()) {
  case TIME:
    return ..
  case VERSION_6:
    return UuidVersion6Strategy.INSTANCE;
  case VERSION_7:
    return UuidVersion7Strategy.INSTANCE;
}
```

Reply:
Well no.  You are missing what is effectively the AUTO, which also happens to  be the `config == null` case.

I'm not following the desire to add a switch just to add a switch.  A switch with so few case options is unuseful imo.  In fact, IntelliJ will even suggest this as a refactoring.

Reply:
Well was trying to flatten the if statement and moved it up as a guard clause for the AUTO. 
https://blog.codinghorror.com/flattening-arrow-code/